docs/topics/whatsnew-eap.md

```diff
@@ -199,23 +199,48 @@ kotlin {
 
 Learn more about the [hierarchical project structure in Kotlin Multiplatform](multiplatform-hierarchy.md).
 
-### Deprecated compatibility with Gradle Java plugins
+### Deprecated compatibility with Kotlin Multiplatform Gradle plugin and Gradle Java plugins
 
-Due to compatibility issues between the Kotlin Multiplatform Gradle plugin and the Gradle plugins 
-[Java](https://docs.gradle.org/current/userguide/java_plugin.html), 
-[Java Library](https://docs.gradle.org/current/userguide/java_library_plugin.html), 
-and [Application](https://docs.gradle.org/current/userguide/application_plugin.html), 
-Kotlin %kotlinEapVersion% introduces a deprecation warning when you apply these plugins in the same project.
-In future Kotlin releases, the warning will be raised to an error.
+In Kotlin %kotlinEapVersion%, we introduce a deprecation warning when you apply the Kotlin Multiplatform Gradle plugin 
+and any of the following Gradle Java plugins within the same project: [Java](https://docs.gradle.org/current/userguide/java_plugin.html),
+[Java Library](https://docs.gradle.org/current/userguide/java_library_plugin.html), and [Application](https://docs.gradle.org/current/userguide/application_plugin.html).
+This warning also appears when another Gradle plugin in your multiplatform project applies a Gradle Java plugin. 
+For example, the [Spring Boot Gradle Plugin](https://docs.spring.io/spring-boot/gradle-plugin/index.html) automatically
+applies the Application plugin.
 
-If you want to use both the Kotlin Multiplatform Gradle plugin in combination with these Gradle plugins for Java in your multiplatform project, 
-we recommend that you:
+We've added this deprecation warning due to fundamental compatibility issues between Kotlin Multiplatform's project model
+and Gradle's Java ecosystem plugins. The problem is that Gradle's Java ecosystem plugins currently don't take into account
+that other plugins may:
+
+* Also publish or compile for the JVM target in a different way than the Java ecosystem plugins.
```

Comment:
```suggestion
* Publish or compile for the JVM target in a different way than the Java ecosystem plugins.
```

Sounds redundant to me, but YMMV.

Reply:
I think it's worth keeping because it's about having two plugins trying to publish for a single JVM target simultaneously :)